packages/wallet/src/base-unlocked-wallet.ts

```diff
@@ -47,7 +47,7 @@ export class BaseWalletUnlocked extends Account {
    * @returns Promise<string> - Signature a ECDSA 64 bytes
    */
   async signMessage(message: string): Promise<string> {
-    return this.signer().sign(hashMessage(message));
+    return Promise.resolve(this.signer().sign(hashMessage(message)));
```

Comment:
I believe they are the same thing. But TS shows a little warning on the second one.

![Screenshot from 2023-03-21 15-08-56](https://user-images.githubusercontent.com/30977845/226702359-8da62736-f775-4882-a19a-b16082914580.png)

Reply:
It's because, in these cases, there is no need to `await` for anything, and the method consumer should do the awaiting. If you keep it like it was before, does it raises warnings? It should be perfectly fine.

```ts
return this.signer().sign(hashMessage(message));
```